presto-accumulo/src/main/java/com/facebook/presto/accumulo/io/AccumuloRecordCursor.java

```diff
@@ -0,0 +1,584 @@
+                IteratorSetting ore = OrFilter.orFilters(priority.getAndIncrement(),
+                        colSettings.toArray(new IteratorSetting[0]));
+                allSettings.add(ore);
+            } // else there are no settings and we can move on
+        }
+
+        // If there is only one iterator, then just use that
+        if (allSettings.size() == 1) {
+            this.scan.addScanIterator(allSettings.get(0));
+        }
+        else if (allSettings.size() > 0) {
+            // Else, and all settings together and add those
+            this.scan.addScanIterator(AndFilter.andFilters(1, allSettings));
+        }
+    }
+
+    /**
+     * Gets settings for a NullRowFilter based on the column constraint
+     *
+     * @param col Column constraint
+     * @param priority Priority of this setting, which is arbitrary in the long run since only one
+     * iterator is set.
+     * @return Iterator settings
+     */
+    private IteratorSetting getNullFilterSetting(AccumuloColumnConstraint col,
+            AtomicInteger priority)
+    {
+        String name = String.format("%s:%d", col.getName(), priority.get());
+        return new IteratorSetting(priority.getAndIncrement(), name, NullRowFilter.class,
+                NullRowFilter.getProperties(col.getFamily(), col.getQualifier()));
+    }
+
+    /**
+     * Gets settings for a SingleColumnValueFilter based on the given column constraint and range.
+     *
+     * @param col Column constraint
+     * @param r Presto range to retrieve values for the column filter
+     * @param priority Priority of this setting, which is arbitrary in the long run since only one
+     * iterator is set.
+     * @return Iterator setting, or null if the Range is all values
+     */
+    private IteratorSetting getFilterSettingFromRange(AccumuloColumnConstraint col, Range r,
+            AtomicInteger priority)
+    {
+        if (r.isAll()) {
+            // [min, max]
+            return null;
+        }
+        else if (r.isSingleValue()) {
+            // value = value
+            return getIteratorSetting(priority, col, CompareOp.EQUAL, r.getType(),
+                    r.getSingleValue());
+        }
+        else {
+            if (r.getLow().isLowerUnbounded()) {
+                // (min, x] WHERE x < 10
+                CompareOp op = r.getHigh().getBound() == Bound.EXACTLY ? CompareOp.LESS_OR_EQUAL
+                        : CompareOp.LESS;
+                return getIteratorSetting(priority, col, op, r.getType(), r.getHigh().getValue());
+            }
+            else if (r.getHigh().isUpperUnbounded()) {
+                // [(x, max] WHERE x > 10
+                CompareOp op = r.getLow().getBound() == Bound.EXACTLY ? CompareOp.GREATER_OR_EQUAL
+                        : CompareOp.GREATER;
+                return getIteratorSetting(priority, col, op, r.getType(), r.getLow().getValue());
+            }
+            else {
```

Comment:
You're right. It's not only not generating any splits but it's not even calling Metadata's `getTableLayouts`, to let you play with constraints.